lib/logstash_sidekiq_logger/formatter.rb

```diff
@@ -0,0 +1,111 @@
+module LogstashSidekiqLogger
+  class Formatter < Logger::Formatter
+    def call(_severity, _time, _program_name, message)
+      store_message_failure_metadata(message)
+
+      performed_job = RequestStore.store[:performed_job]
+      return unless performed_job && message_needs_logging?(message)
+
+      data = extract_log_data(performed_job, message)
+      event = LogStash::Event.new(data)
+
+      RequestStore.clear!
+
+      # Is there a better way to make the logs appear on new lines??
+      "\n" + event.to_json
+    end
+
+  private
+
+    def store_message_failure_metadata(message)
+      return if message !~ /^fail/
+
+      extract_failure_duration(message)
+    end
+
+    def extract_failure_duration(message)
+      return if message !~ /^fail/
```

Comment:
This method appears to only be used in one place (`store_message_failure_metadata`, directly above), which *also* ensures that the message starts with `fail`. Maybe merge the two?